File: dart/algorithms/searching/linear_search.dart
// linear_search.dart

int linearSearch(List<int> array, int target) {
  for (int i = 0; i < array.length; i++) {
    if (array[i] == target) {
      return i; // Return the index if the target is found
    }
  }
  return -1; // Return -1 if the target is not found
}

void main() {
  List<int> array = [2, 4, 6, 8, 10, 12, 14];
  int target = 8;

  int result = linearSearch(array, target);

  if (result != -1) {
    print("Element $target found at index $result.");
  } else {
    print("Element $target not found in the array.");
  }
}

File: go/algorithms/searching/linear_search.go
// linear_search.go

package main

import "fmt"

func linearSearch(arr []int, target int) int {
    for i, value := range arr {
        if value == target {
            return i
        }
    }

    return -1
}

func main() {
    array := []int{10, 4, 6, 8, 2, 7, 1}
    target := 6

    result := linearSearch(array, target)

    if result != -1 {
        fmt.Printf("Element %d found at index %d\n", target, result)
    } else {
        fmt.Printf("Element %d not found in the array\n", target)
    }
}
